### ISSUE TITLE
Return access token and secret for Twitter login

### ISSUE THREAD

**Issue by rishimohan** [supabase/supabase-js#230]
# Feature request

## Is your feature request related to a problem? Please describe.

Not exactly a problem, I'm building a full-fledged Twitter app for web and using Supabase as database right now. The issue is I'm not using Supabase's auth for Twitter login, so basically there's loophole to writing to the tables with bearer token in my app. I want to use Supabase's auth with auth policy to fix this situation.

The issue with Twitter sign-in with Supabase is that it doesn't return user's access token and secret in session or anything. I want these user credentials so I can load their feed and make tweets in their behalf. 

## Describe the solution you'd like

It would be great if we can Twitter oauth can return user's access token and secret as well.

## Describe alternatives you've considered

-

## Additional context

I have tried Supabase Twitter auth, I've checked it returns the bearer token and access token, but it doesn't return access secret. Both user's access secret and token are needed to make request on user's behalf, I hope it's something Supabase can add to Twitter auth. Thanks,

**Comment by adam-beck** [comment]
Is this related to https://github.com/supabase/supabase-js/issues/145

**Comment by rishimohan** [comment]
@adam-beck I suppose no, this issue is a proposal if we can return the user's access keys when logged in through Twitter so that requests can be made to Twitter API on user's behalf.

**Comment by adam-beck** [comment]
Yeah, sorry. After reading #145 closer, I realize they are asking for something different. In that ticket, the access token for the OAuth Provider is already in possession, and they want to use it directly with Supabase. Here, you're asking for somewhat of the opposite.

My bad

**Comment by rishimohan** [comment]
@adam-beck No issues :)

**Comment by DarrylBrooks97** [comment]
Yoo this is the exact issue I'm running into. This would be a great addition to the sdk

**Comment by eramdam** [comment]
I could really use this too. Does anyone know of a possible workaround? 😅

**Comment by rishimohan** [comment]
@eramdam I had to fall back to using Next auth for my Next.js project(Kizie.co). I had a call with one of Supabase members some time back and he mentioned that they'll be working on this issue.

**Comment by eramdam** [comment]
> @eramdam I had to fall back to using Next auth for my Next.js project(https://kizie.co). I had a call with one of Supabase members some time back and he mentioned that they'll be working on this issue.

"Next auth" as in [NextAuth.js](https://next-auth.js.org/providers/twitter) or are you referring to something else?

**Comment by rishimohan** [comment]
@eramdam Yep :)

**Comment by tzhf** [comment]
I'm having the same difficulties with the Twitch provider. I really wonder why supabase doesn't provide the access_token in the response. It's needed in so many use cases, without access_token we are limited to get user's info only. I need my user to be able to interact with Twitch with that token.

> I could really use this too. Does anyone know of a possible workaround? 😅
- Dirty workaround for me, (not sure it's the same for Twitch) by intercepting the url containing that token right after the redirection.

**Comment by kirillinoz** [comment]
Are there any updates on this issue? Is it possible to return the access secret in the meantime?

**Comment by awalias** [comment]
I'm going to transfer this to the auth repo
